### PR TITLE
[QA-1727] Add logging.logWriter to list of expected roles

### PR DESCRIPTION
**Pull request by aherbst-broad**
The additional role binding of `logging.logWriter` is causing a test failure. This PR adds the role to the expected list. Given the evident volatility of role bindings, this PR also relaxes the assertion to a warning to prevent future failures.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala [hunk 1]
```diff
@@ -143,13 +143,13 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with AnyFreeSpecLi
               case s if s.endsWith("editor") => binding.getMembers.size() shouldEqual 2
               case s if s.endsWith("genomics.serviceAgent") => binding.getMembers.size() shouldEqual 1
               case s if s.endsWith("lifesciences.serviceAgent") => binding.getMembers.size() shouldEqual 1
+              case s if s.endsWith("logging.logWriter") => binding.getMembers() shouldEqual 1
```
Comment (ungwudik) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: Should `binding.getMembers()` be `binding.getMembers.size()`, or it doesn't matter?

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala [hunk 2]
```diff
@@ -143,13 +143,13 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with AnyFreeSpecLi
...
               case s if s.endsWith("owner") =>
                 binding.getMembers.size() shouldEqual 1
                 binding.getMembers.get(0) should endWith("@terra-kernel-k8s.iam.gserviceaccount.com")
               case s if s.endsWith("pubsub.serviceAgent") => binding.getMembers.size() shouldEqual 1
               case other =>
-                logger.error(s"Extra permission on workspace google project found: ${other}")
-                throw new Exception(s"Extra permission on workspace google project found: ${other}")
```
Comment (ungwudik) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:152]: It seems like this line may still be important, if we want to be certain about which permissions a project should have, or shouldn't have for auth'z reasons?

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala (hunk 1 above)
Comment (aherbst-broad) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: Gah, good catch, fixed.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala (hunk 2 above)
Comment (aherbst-broad) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:152]: Sure. The issue is that I am unaware of any authoritative list of role bindings to expect and this test is prone to breakage when the effective list is changed via external services (which is what happened here). The somewhat weaker assertion earlier in the test for _at least_ the minimal set of roles is the best compromise I could think of.

Comment (marctalbott) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:152]: I'm questioning whether this test is worth keeping. The match statement is still quite brittle even without this exception. An external service adding a new SA or Sam policy group with an existing role (ie increasing the size of one of the role bindings above) will break the test.

Comment (aherbst-broad) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:152]: @marctalbott that's a fair point. Perhaps we just limit to asserting the set diff is sane (i.e,. the "core" set of role bindings are present) and don't even bother with the case match?

Comment (aherbst-broad) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:152]: ok, I think I am sold on removing the case match given it's fragility and proven ability to break the build. I think a check of this nature is still valuable, I will drop a ticket to remediate.

Comment (aherbst-broad) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:152]: @ungwudik does this work for you?

Comment (ungwudik) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:152]: Yes, that works.

**Review by ungwudik — APPROVED** [review]

**Review by marctalbott — APPROVED** [review]

**Pull request merged**